supertokens_python/supertokens.py

```diff
@@ -122,13 +116,19 @@ def __init__(
         mode: Union[Literal["asgi", "wsgi"], None],
     ):
         self.app_name = app_name
-        self.api_gateway_path: NormalisedURLPath = NormalisedURLPath(api_gateway_path)
-        self.api_domain: NormalisedURLDomain = NormalisedURLDomain(api_domain)
-        self.website_domain: NormalisedURLDomain = NormalisedURLDomain(website_domain)
-        self.api_base_path: NormalisedURLPath = self.api_gateway_path.append(
+        self.api_gateway_path = NormalisedURLPath(api_gateway_path)
+        self.api_domain = NormalisedURLDomain(api_domain)
+        self.website_domain = NormalisedURLDomain(website_domain)
+        self.top_level_api_domain = get_top_level_domain_for_same_site_resolution(
```

Comment:
- Update the session utils to use this instead of calling `get_top_level_domain_for_same_site_resolution` again
- move `get_top_level_domain_for_same_site_resolution` out of the session recipe

Reply:
Done